smpclient/__init__.py

```diff
@@ -97,32 +98,53 @@ async def __aenter__(self) -> 'SMPClient':
     async def __aexit__(self) -> None:
         await self.disconnect()
 
-    def _maximize_packet(self, request: ImageUploadWrite, image: bytes) -> ImageUploadWrite:
-        """Given an `ImageUploadWrite` with empty `data`, return the largest packet possible."""
+    @staticmethod
+    def _cbor_integer_size(integer: int) -> int:
+        """CBOR integers are packed as small as possible."""
+
+        return 0 if integer < 24 else 1 if integer < 0xFF else 2 if integer < 0xFFFF else 4
```

Comment:
Maybe notify that 24 here is a significance for short field encoding? otherwise it looks like a magic number